### DISCUSSION TITLE
How to Set Up PWM Input Capture Mode in Zephyr for STM32G0B0?

### DISCUSSION THREAD

**Discussion by arturodrt** [zephyrproject-rtos/zephyr#79885] [Q&A]
Hello everyone,

I'm currently working on a project using the STM32G0B0 with the Zephyr RTOS, and I need to configure PWM input capture mode. I'm looking for guidance or clear examples on how to achieve this.

Specifically, I have the following questions:

- What are the necessary steps to configure the PWM input capture mode in the device tree? Are there specific properties I need to set?

- Is there a sample code snippet that demonstrates how to initialize and use PWM input capture in a Zephyr application?

So far I am using this pwm_loopback example https://github.com/sofarocean/zephyr/tree/sofar/main/tests/drivers/pwm/pwm_loopback
and I am trying to follow https://docs.zephyrproject.org/apidoc/latest/group__pwm__interface.html 

Thanks in advance!

**Comment by erwango** [comment]
@gautierg-st maybe ?
